ftw/upgrade/utils.py

```diff
@@ -244,3 +245,27 @@ def get_tempfile_authentication_directory(directory=None):
         raise ValueError('{0} has an invalid owner.'.format(auth_directory))
 
     return auth_directory
+
+
+class StartsWithLogFilter(logging.Filter):
+    """Filter messages that start with criteria."""
+
+    def __init__(self, criteria):
+        self.criteria = criteria
+
+    def filter(self, record):
+        return not record.getMessage().startswith(self.criteria)
+
+
+@contextmanager
+def log_silencer(logger_name, criteria):
+    """Prevents messages that start with `criteria` from being logged to the
+    logger that is registered as `logger_name`.
+    """
+    log = logging.getLogger(logger_name)
+    filt = StartsWithLogFilter(criteria)
+    log.addFilter(filt)
+
+    yield
```

Comment:
I think we should ensure that the filter is removed too when an error happens; otherwise the filter would stay in the instance until is rebooted.
```python
try:
    yield
finally:
    log.removeFilter(filt)
```